src/utils/index.ts

```diff
@@ -0,0 +1,59 @@
+import type { Data, Inputs, AttributeVal, HtmlAttributes } from '../types';
+
+// Add all required search params with user inputs as values
+function formatUrl(url: string, params: string[], args: Inputs) {
+  if (!params || !args) return url;
+
+  let newUrl = new URL(url);
```

Comment:
Nit, doesn't look like newUrl changes assignment

```suggestion
  const newUrl = new URL(url);
```